### ISSUE TITLE
Eventsourced.persist misbehaves invoked from within RecoveryCompleted

### ISSUE THREAD

**Issue by minisaw** [akka/akka#22609]
akka.persistence.Eventsourced.persist javadoc documentation states:

_It is guaranteed that no new commands will be received by a persistent actor between a call to `persist` and the execution of its `handler`._

However, in case an event is persisted from within the RecoveryCompleted message handler, the statement above does not hold. The following sequence of events happens:

1. upon creation, the actor's receiveRecover handler gets the RecoveryCompleted message.
2. in RecoveryCompleted message handler, persist an event E
3. send a message X to the actor
4. the actor's receiveCommand handler receives the message X, processes it
5. only then the E's handler is invoked

While according to the specification, bullet number 4 must have occurred only after bullet 5.

A test application showcasing the issue is available at https://github.com/minisaw/akka-persistence-test

also another mention of the same problem:
https://stackoverflow.com/questions/36649466/akka-persist-on-recovery-completed-updates-state-after-first-message

**Comment by patriknw** [comment]
@minisaw Could this be a duplicate of https://github.com/akka/akka/issues/21736 ?

**Comment by minisaw** [comment]
it appears to be.

**Comment by minisaw** [comment]
there's one more quirk, though: the event handler will not be invoked at all in case no message is sent to the actor. so the problem is not only in that the commands are not stashed (as #21736 states).

**Comment by patriknw** [comment]
so https://github.com/akka/akka/pull/21893 is not covering everything?

**Comment by minisaw** [comment]
works in 2.4.18, closing
